tests/constraints.txt

```diff
@@ -4,3 +4,6 @@ pep517==0.9.1
 toml==0.10.0
 typing==3.5.3.0;python_version < "3"
 virtualenv==20.0.35;python_version < "3"
+trampolim==0.0.2;python_version < "3.7"
```

Comment:
This looks like a dependency that might be difficult to satisfy, e.g. for distros.  Can we have an in-tree backend which implements a barebones `build_wheel` hook for testing purposes?  Something like:

```patch
diff --git a/tests/packages/test-no-prepare/prepareless_backend.py b/tests/packages/test-no-prepare/prepareless_backend.py
new file mode 100644
index 0000000..44e932d
--- /dev/null
+++ b/tests/packages/test-no-prepare/prepareless_backend.py
@@ -0,0 +1,27 @@
+import os.path
+import zipfile
+
+
+def build_wheel(wheel_directory, config_settings=None, metadata_directory=None):
+    with zipfile.ZipFile(
+        os.path.join(wheel_directory, 'test_no_prepare-1.0.0-py3-none-any.whl'),
+        'w',
+    ) as wheel:
+        wheel.writestr(
+            'test_no_prepare-1.0.0.dist-info/METADATA',
+            '''\
+Metadata-Version: 2.2
+Name: test-no-prepare
+Version: 1.0.0
+'''
+        )
+        wheel.writestr(
+            'test_no_prepare-1.0.0.dist-info/WHEEL',
+            '''\
+Wheel-Version: 1.0
+Generator: build
+Root-Is-Purelib: true
+Tag: py3-none-any
+'''
+        )
+    return wheel.filename
diff --git a/tests/packages/test-no-prepare/pyproject.toml b/tests/packages/test-no-prepare/pyproject.toml
index 23bccd8..8795521 100644
--- a/tests/packages/test-no-prepare/pyproject.toml
+++ b/tests/packages/test-no-prepare/pyproject.toml
@@ -1,7 +1,4 @@
 [build-system]
-build-backend = 'trampolim'
-requires = ['trampolim']
-
-[project]
-name = 'test-no-prepare'
-version = '1.0.0'
+build-backend = 'prepareless_backend'
+requires = []
+backend-path = ['.']

```

Reply:
I changed this to setuptools and used an intree backend that just does `from setuptools.build_meta import build_sdist, build_wheel`.